src/lab12.py

```diff
@@ -1361,10 +1361,21 @@ def handle_quit(self):
 
 if __name__ == "__main__":
     import sys
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Chapter 12 code')
+    parser.add_argument("--url", default=2, type=str, required=True,
```

Comment:
```suggestion
    parser.add_argument("url", type=str, required=True,
```

The name should not have dashes if it's a positional (not named) argument, and `default` describes a default value, not the position.

Reply:
Apparently the -- is needed...seems to work as desired now.

Examples here use it: https://docs.python.org/3/library/argparse.html